frontend/src/components/ui/Header.tsx

```diff
@@ -24,11 +24,13 @@ export function Header() {
                 <Button variant="ghost">Collection</Button>
               </Link>
             </NavigationMenuLink>
-            <NavigationMenuLink asChild>
-              <Link to="/trade">
-                <Button variant="ghost">Trade</Button>
-              </Link>
-            </NavigationMenuLink>
+            {user && (
```

Comment:
I would still show the tab so before registering you at least know the features exists and you can already look at it. I did the same for the collection page (and want to do the same for the overview page). The idea is you can view all the features of the app, but you need to register to track cards.